crates/blockifier/src/state/visited_pcs.rs

```diff
@@ -0,0 +1,94 @@
+use std::collections::hash_map::{Entry, IntoIter, Iter};
+use std::collections::{HashMap, HashSet};
+use std::fmt::Debug;
+
+use starknet_api::core::ClassHash;
+
+use super::state_api::State;
+
+/// This trait is used in `CachedState` to record visited pcs of an entry point call.
+pub trait VisitedPcs
+where
+    Self: Default + Debug,
+{
+    /// This is the type which contains visited program counters.
+    type T: Clone;
```

Comment:
Probably best to give this a more descriptive name than `T`. `PCContainer` or something.